app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -131,6 +133,16 @@ def leave_membership
     end
   end
 
+  def request_details
+    mod_memberships = ChatChannelMembership.where(user_id: current_user.id, role: "mod", status: "active")
+    user_chat_channels = mod_memberships.map(&:chat_channel)
```

Comment:
You could get these chat channel entirely using SQL
```ruby
user_chat_channels = ChatChannel.includes(:chat_channel_memberships).where(chat_channel_memberships: { user_id: current_user.id, role: "mod", status: "active" })
```

I am also noticing we are doing a lot of channel membership searches by status and role, it might be helpful to make some scopes for those to consolidate and refactor some of the logic.